src/content/operationDashboard/DashboardView.tsx

```diff
@@ -0,0 +1,137 @@
+import { Box, FormControl, MenuItem, SelectChangeEvent, styled, Typography } from "@mui/material";
+import { isEqual } from "lodash";
+import { FC, memo, useEffect, useRef, useState } from "react";
+import { useSearchParams } from "react-router-dom";
+import {
+  createEmbeddingContext,
+  DashboardExperience,
+  FrameOptions,
+} from "amazon-quicksight-embedding-sdk";
+import StyledSelect from "../../components/StyledFormComponents/StyledSelect";
+import SuspenseLoader from "../../components/SuspenseLoader";
+import bannerSvg from "../../assets/banner/submission_banner.png";
+
+export type DashboardViewProps = {
+  url: string;
+  currentType: string;
+  loading: boolean;
+};
+
+const StyledViewHeader = styled(Box)({
+  background: `url(${bannerSvg})`,
+  backgroundSize: "cover",
+  backgroundPosition: "center",
+  width: "100%",
+  height: "296px",
+  display: "flex",
+  justifyContent: "center",
+  alignItems: "center",
+  marginBottom: "-130px",
+  marginTop: "0px",
+});
+
+const StyledFormControl = styled(FormControl)({
+  display: "flex",
+  flexDirection: "row",
+  alignItems: "center",
+  gap: "15px",
+  width: "300px",
+  marginBottom: "auto",
+  marginTop: "57px",
+});
+
+const StyledInlineLabel = styled("label")({
+  padding: 0,
+  fontWeight: "700",
+});
+
+const StyledFrameContainer = styled(Box)({
+  borderRadius: "6px",
+  border: "1px solid #E0E0E0",
+  background: "#fff",
+  position: "relative",
+  margin: "0 auto",
+  marginBottom: "57px",
+  maxWidth: "calc(100% - 64px)",
+  boxShadow:
+    "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
+});
+
+const StyledPlaceholder = styled(Typography)({
+  margin: "100px auto",
+  textAlign: "center",
+  userSelect: "none",
+  fontSize: "16px",
+  color: "#5C5C5C",
+});
+
+/**
+ * The view for the OperationDashboard component.
+ *
+ * @param {DashboardViewProps} props The props for the component.
+ * @returns {JSX.Element} The OperationDashboard component.
+ */
+const DashboardView: FC<DashboardViewProps> = ({
+  url,
+  currentType,
+  loading,
+}: DashboardViewProps) => {
+  const [, setSearchParams] = useSearchParams();
+  const [embeddedDashboard, setEmbeddedDashboard] = useState<DashboardExperience>(null);
+  const dashboardElementRef = useRef<HTMLDivElement>(null);
+
+  const handleDashboardChange = (e: SelectChangeEvent) => {
+    setSearchParams({ type: e.target.value });
+    dashboardElementRef.current.innerHTML = "";
+    setEmbeddedDashboard(null);
+  };
+
+  const createEmbed = async () => {
```

Comment:
This is mostly for development QOL, but could you add a guard to check if embeddedDashboard already is populated, if it is then return? The reason for this is when you make a change on this file and React hot reloads the page, then it will try to embed a duplicate iframe into the project which always fails. You can test this by loading the page, adding a console log, then viewing the page again without reloading. 

![Screenshot 2024-08-29 at 12 18 09 PM](https://github.com/user-attachments/assets/e8310229-05b6-464b-ad47-23e8d56e0b93)

Reply:
I updated this to make the dashboard slightly more stable. This will have to be re-evaluated once we have additional dashboards, as I think we would need to rebuild the embeddedDashboard each time the selectbox changes. For now, this change should work fine though. e49ac8e